README.md

```diff
@@ -102,6 +103,23 @@ Succinct help about any command or subcommand is available via the `-h, --help`
 flag. Verbose help about any command or subcommand is available via the help
 argument, e.g. `fastly help service`.
 
+## Bash/ZSH Shell Completion
+The CLI can generate completions for all commands, subcommands and flags.
+
+By specifying `--completion-bash` as the first argument, the CLI will show possible subcommands. By ending your argv with `--`, hints for flags will be shown.
+
+### Installation by bash_profile
+
+To install the completions source them in your `bash_profile` (or equivalent):
+```
+eval "$(fastly --completion-script-bash)"
+```
+
+Or for ZSH in your `zshrc`:
+```
+eval "$(fastly --completion-script-zsh)"
+```
+
```

Comment:
I find this a tad confusing - what's the difference between `--completion-bash` and `--completion-script-bash`?  Does installing it in the profile avoid the need to use the `--completion-bash` flag?  When would you want to do one vs the other?  And when we say autocomplete do we mean tab-based autocomplete?  If not how do I trigger the autocompletion (I know this is a shell specific thing but you're providing instructions for both popular shells and I imagine the resulting UX would be similar?)

Reply:
Good questions. 

The `--completetion-script-bash` generates the shell script that you'd source in your bash profile via `eval "$(fastly --completion-script-bash)"` or similar, and tells your shell how to load the tab autocompletions.
```
_fastly_bash_autocomplete() {
    local cur prev opts base
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    opts=$( ${COMP_WORDS[0]} --completion-bash ${COMP_WORDS[@]:1:$COMP_CWORD} )
    COMPREPLY=( $(compgen -W "${opts}" -- ${cur}) )
    return 0
}
complete -F _fastly_bash_autocomplete fastly
```

The `--completion-bash` actually provides the completions in a newline delimited list. Such as:
```
create
list
describe
update
delete
```

So TL;DR the script tells your shell how to invoke the CLI with the completion flag to get the list of tab completions.

Reply:
That's great, thanks.  Do you think people other than me would benefit from that context?  If so, you could update the README text.